how-to/workspace-platform-starter/client/types/module/shapes/favorite-shapes.d.ts

```diff
@@ -0,0 +1,198 @@
+import type { PlatformStorageMetadata } from "./platform-shapes";
+/**
+ * Names for all the favorite types.
+ */
+export type FavoriteTypeNames = "app" | "workspace" | "page" | "query";
+/**
+ * Options for the favorite provider.
+ */
+export interface FavoriteProviderOptions {
+	/**
+	 * The icon that should be used if you want to indicate this is a favorite action
+	 */
+	favoriteIcon: string;
+	/** The icon to use to indicate that this favorite can be unset */
+	unfavoriteIcon: string;
+	/**
+	 * What commands should integrations check for if they intent to support the display of favorites
+	 */
+	favoriteCommand?: string;
+	/**
+	 * The connection provider can have actions registered against it from the platform. This provides a default list of
+	 * actions that connections should be able to use if actions are enabled for that connection.
+	 */
+	supportedFavoriteTypes?: FavoriteTypeNames[];
+}
+/**
+ * When an entry is made it represents a type supported by this platform. This can be used to lookup and launch the thing this entry refers to.
+ */
+export interface FavoriteEntry {
+	/**
+	 * A unique guid to represent this favorite entry so that it can be updated or removed
+	 */
+	id: string;
+	/**
+	 * The id for the favorite type this entry represents
+	 */
+	typeId: string;
+	/**
+	 * What type of favorite entry does this entry represent
+	 */
+	type: FavoriteTypeNames;
+	/**
+	 * The timestamp for the entry.
+	 */
+	timestamp: Date;
+	/**
+	 * Does this favorite have a suggested label that can be used to avoid a lookup
+	 */
+	label?: string;
+	/**
+	 * Does this favorite have a suggested icon that can be used to avoid a lookup
+	 */
+	icon?: string;
+}
+/**
+ * Info to return to interested parties to help them support favorites
+ */
+export interface FavoriteInfo {
+	/**
+	 * The path to an icon that can be used to indicate the ability to favorite
+	 */
+	favoriteIcon?: string;
+	/**
+	 * The path to an icon that can be used to indicate the ability to remove this favorite
+	 */
+	unfavoriteIcon?: string;
+	/**
+	 * A command that supporting modules should listen for if they are to display favorites that fall under them
+	 */
+	command?: string;
+	/**
+	 * What types of favorite item are supported on this platform
+	 */
+	enabledTypes?: FavoriteTypeNames[];
+	/**
+	 * Is favorite support initialized on this platform.
+	 */
+	isInitialized: boolean;
+}
+/**
+ * A client that can be used to provide access to some or all of the favorite functionality
+ */
+export interface FavoriteClient {
+	/**
+	 * The ability to request all (or some if by type) of the saved favorites
+	 * @param byType the type of saved favorite you are looking for
+	 * @returns An array of saved favorites or an empty array if it was unable to get any back
+	 */
+	getFavorites?: (byType: FavoriteTypeNames) => Promise<FavoriteEntry[]>;
```

Comment:
All the methods probably shouldnt be optional, either the whole client is available or its not

Reply:
changed it. getInfo, getFavorites and getFavorite are now always there but add and remove is optional as we might not want to give everything that option